ExchangeOnline/Export-EXOMailboxSizes.ps1

```diff
@@ -279,28 +319,50 @@ Write-Verbose "There are $mailboxCount mailboxes"
 
 if ($mailboxCount -eq 0)
 {
-    return 'There are no mailboxes found using the supplied filters'
+    throw 'There are no mailboxes found using the supplied filters'
 }
 
 #  Loop through the list of mailboxes and output the results to the CSV
 Write-Verbose 'Beginning loop through all mailboxes'
 foreach ($mailbox in $mailboxes)
 {
-    if (!$PSCmdlet.MyInvocation.BoundParameters['Verbose'].IsPresent)
+    # if InputCSV is specified, match against mailbox list
+    if ($InputCSV)
```

Comment:
I am still wondering if there is a good way to construct an oPath filter and get Exchange to do the hard work, only returning the mailboxes we care about!